src/generate/register.rs

```diff
@@ -510,6 +510,93 @@ fn render_register_mod_debug(
     Ok(r_debug_impl)
 }
 
+#[derive(Clone, Copy, Debug)]
+pub enum EV<'a> {
+    New(&'a EnumeratedValues),
+    Derived(&'a EnumeratedValues, &'a EnumPath),
+}
+
+impl<'a> EV<'a> {
+    fn values(&self) -> &EnumeratedValues {
+        match self {
+            Self::New(e) | Self::Derived(e, _) => e,
+        }
+    }
+}
+
+impl<'a> From<&'a (EnumeratedValues, Option<EnumPath>)> for EV<'a> {
+    fn from(value: &'a (EnumeratedValues, Option<EnumPath>)) -> Self {
+        match value.1.as_ref() {
+            Some(base) => Self::Derived(&value.0, &base),
+            None => Self::New(&value.0),
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum RWEnum<'a> {
+    ReadAndWriteEnum(EV<'a>),
+    ReadEnumWriteEnum(EV<'a>, EV<'a>),
+    ReadEnumWriteRaw(EV<'a>),
+    ReadRawWriteEnum(EV<'a>),
+    ReadEnum(EV<'a>),
+    ReadRaw,
+    WriteEnum(EV<'a>),
+    WriteRaw,
+    ReadRawWriteRaw,
+}
```

Comment:
I think it would be more ergonomic to represent this as

```rs
enum RWEnum<'a> {
   Read(ReadEnum<'a>),
   Write(WriteEnum<'a>),
   ReadWrite(ReadEnum<'a>, WriteEnum<'a>),
}

enum ReadEnum<'a> {
    Enum(EV<'a>)
    Raw,
}

enum WriteEnum<'a> {
    Enum(EV<'a>)
    Raw,
}
```

Reply:
Done.